modules/scaleform.lua

```diff
@@ -0,0 +1,224 @@
+---@class Scaleform : OxClass
+---@field name string
+---@field scaleform number
+---@field draw boolean
+---@field target number
+---@field targetName string
+---@field handle number
+local Scaleform = lib.class('Scaleform')
+
+---@param details table | string
+---@return nil
+---@description Create a new scaleform class
+function Scaleform:constructor(details)
+    local detailProvided = type(details) == "table"
+    self.name = detailProvided and details.name or details -- Set the name
+
+    local scaleform = lib.requestScaleformMovie(self.name) -- Request the scaleform movie
+
+    if not scaleform then -- If the scaleform is nil
+        return error(('Failed to request scaleform movie - [%s]'):format(self.name)) -- Error the failed scaleform request
+    end
+
+    self.handle = scaleform -- Set the scaleform handle
+    self.draw = false -- Set the draw to false
+    self.fullScreen = detailProvided and details.fullScreen or true -- Set default for full screen
+
+    if detailProvided then
+        self.x = details.x
+        self.y = details.y
+        self.width = details.width
+        self.height = details.height
+
+        if details.renderTarget then
+            self:renderTarget(details.renderTarget.name, details.renderTarget.model)
+        end
+    end
+end
+
+local function convertArgs(argsTable)
+    for i=1, #argsTable do -- loop through the args
+        local arg = argsTable[i] -- Set the value to the current arg
+        if type(arg) == 'string' then -- If the type of v is a string
+            ScaleformMovieMethodAddParamPlayerNameString(arg) -- Add the player name string
+        elseif type(arg) == 'number' then -- If the type of v is a number
+            if math.type(arg) == 'integer' then -- If the math type of v is an integer
+                ScaleformMovieMethodAddParamInt(arg) -- Add the integer
+            else -- If the math type of v is not an integer
+                ScaleformMovieMethodAddParamFloat(arg) -- Add the float
+            end
+        elseif type(arg) == 'boolean' then -- If the type of v is a boolean
+            ScaleformMovieMethodAddParamBool(arg) -- Add the boolean
+        else
+            error(('Unsupported Parameter type [%s]'):format(type(arg))) -- Error unsupported type
+        end
+    end
+end
+
+---@param type string
+---@return boolean | int | string
```

Comment:
```suggestion
---@param type 'bool' | 'int' | 'string'
---@return boolean | integer | string
```
Specific types are appreciated, makes it more readable.